src/Compiler/Microsoft.AspNetCore.Razor.Language/src/CodeGeneration/CodeWriterExtensions.cs

```diff
@@ -543,7 +543,7 @@ public static IDisposable BuildLinePragma(this CodeWriter writer, SourceSpan? sp
         return new LinePragmaWriter(writer, sourceSpan, context, 0, false);
     }
 
-    public static IDisposable BuildEnhancedLinePragma(this CodeWriter writer, SourceSpan? span, CodeRenderingContext context, int characterOffset = 0)
+    public static IDisposable BuildEnhancedLinePragma(this CodeWriter writer, SourceSpan? span, CodeRenderingContext context, int characterOffset = 1)
```

Comment:
❓ why 1 here instead of 0?

Reply:
Also curious why this changed.

Reply:
This was actually a bug. The enhanced line directive spec states that all values are 1 based, so 1 indicated 'no offset' essentially.

Reply:
Do we have a test covering this bug? I'd also rather break out this bugfix in a separate PR, so that it can be reverted/adjusted if necessary. Historically, changing things like this has revealed other subtle bugs, so I'd rather keep the change isolated.

Reply:
Sounds good. I'll revert the default here to `0` and manually pass `1` where we're relying on the default value. I'll open a separate PR in main to update the default value and add a test.